File: src/main/java/com/wowo/wowo/domain/user/valueobjects/Email.java
package com.wowo.wowo.domain.user.valueobjects;

/**
 * Email value object with validation
 */
public record Email(String value) {
    
    public Email {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException("Email cannot be null or empty");
        }
        if (!isValidEmail(value.trim())) {
            throw new IllegalArgumentException("Invalid email format");
        }
    }
    
    public static Email of(String email) {
        return new Email(email.trim().toLowerCase());
    }
    
    private static boolean isValidEmail(String email) {
        return email.contains("@") && email.contains(".");
    }
    
    @Override
    public String toString() {
        return value;
    }
}

File: src/main/kotlin/com/wowo/wowo/shared/valueobject/Email.kt
package com.wowo.wowo.shared.valueobject

import com.wowo.wowo.shared.domain.ValueObject

/**
 * Value Object representing an email address
 */
data class Email(val value: String) : ValueObject {
    init {
        require(value.isNotBlank()) { "Email cannot be blank" }
        require(isValidEmail(value)) { "Invalid email format: $value" }
    }

    private fun isValidEmail(email: String): Boolean {
        val emailRegex = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$".toRegex()
        return emailRegex.matches(email)
    }

    override fun toString(): String = value
}

